### ISSUE TITLE
[Bug] Parameters not in order

### ISSUE THREAD

**Issue by toni-moreno** [rs/zerolog#155]
Hi We are working with zerolog to create a kind of HTTP access.log and as mentioned here https://github.com/rs/zerolog/issues/50#issuecomment-380246612 I need the order  I use in the code.

Here the code.

https://github.com/toni-moreno/influxdb-srelay/blob/0e7f56f7c6e3f626773fa17d88b33ad20b453fda/relay/http_middlewares.go#L69-L84

````golang
		h.acclog.Info().
			Str("trace-route", rc.TraceRoute.String()).
			Str("referer", r.Referer()).
			Str("url", r.URL.String()).
			Int("write-size", rc.RequestSize).
			Int("write-points", rc.RequestPoints).
			Int("returnsize", rc.SentDataLength).
			Dur("duration_ms", time.Since(rc.InputTime)).
			Dur("bk_duration_ms", time.Since(rc.BackendTime)).
			Dur("latency_ms", rc.BackendTime.Sub(rc.InputTime)).
			Int("status", rc.SentHTTPStatus).
			Str("method", r.Method).
			Str("user", utils.GetUserFromRequest(r)).
			Str("source", r.RemoteAddr).
			Str("user-agent", r.UserAgent()).
			Msg("")
````

And here the output with the consolewritter

```
2019-06-01 07:44:20 INF  bk_duration_ms=30.448107 duration_ms=113.510554 latency_ms=83.075333 method=POST referer= returnsize=0 source=127.0.0.1:45078 status=204 trace-route="http:example-http-influxdb2> rt:telegraf> decode:ILP> rule:rename_mem_measurement_to_mem_2> rule:to_namespace_metrics> rule:route_all_data_to_cluster_linux> " url=/write?db=telegraf user=admin user-agent=telegraf write-points=23 write-size=4770
```

As you can see there is no order in the output

**Comment by rs** [comment]
The json out is ordered as written in the code. The ConsoleWriter does not keep this order because it does umarshaling of the json to format it.

**Comment by toni-moreno** [comment]
Hi @rs thank you for the fast response.

There is any workaround to force an specific order also with ConsoleWriter?

**Comment by rs** [comment]
We would need to change the wait the json is unmarshaled. Possible but requires a patch.

**Comment by TommyCpp** [comment]
Hi I came across this issue and I think maybe we can add a function in `ConsoleWriter` that allows user to customize the order of output field.

**Comment by toni-moreno** [comment]
Hi @TommyCpp this sounds great!

**Comment by BenjamenMeyer** [comment]
Ordering by the JSON keys is important as it allows one to more easily parse/read through the logs since fields are in the same order. Having random ordering of the key-values make it hard to read when going through logs.

If your only goal is to load into an ELK stack or similar then ordering may not matter as much. However, developers often need to be able to read log files locally and this makes a huge difference.

Having ordering as an option that can be enabled/disabled would still allow for the increased speed where folks don't care, and the negligible slowdown when they do.

**Comment by WQPhello** [comment]
I completely agree with your point of view. Formatted logs can help developers, testers, and operations personnel quickly read them without the need for complex searching and eye scanning.

**Comment by madkins23** [comment]
Just submitted PR #550.

**Comment by heaven** [comment]
Any chance for this to get merged?

**Comment by phuslu** [comment]
I think it's hardly/impossible to fix on top of stdlib, that's why I develop a home made json parser based on gjson to implement ordering text fields as logging statement. https://github.com/phuslu/log/blob/master/formatter.go
